tests/rptest/tests/remote_label_test.py

```diff
@@ -148,3 +167,115 @@ def clusters_report_identical_hwms():
         assert len(
             out_lines
         ) == num_records, f"output has {len(out_lines)} lines: {out}"
+
+    @cluster(num_nodes=2,
+             log_allow_list=[
+                 re.compile("No such file or directory.*cloud_storage_cache")
+             ])
+    @matrix(cloud_storage_type=get_cloud_storage_type())
+    def test_share_bucket_concurrent_consume(self, cloud_storage_type) -> None:
+        """
+        - cluster 1 creates topic_a
+        - cluster 2 creates topic_a
+        - cluster 1 produces to topic_a with pattern X
+        - cluster 2 produces to topic_a with pattern Y
+        - both clusters continue producing until there are many segments in
+          cloud, and wait for local data to be reclaimed
+        - both clusters verify their workloads independently
+        - both clusters unmount and then mount the other clusters' topic
+        - verify the swapped workloads
+        """
+        topic_name = "topic-a"
+        c1 = self.redpanda
+        c2 = self.start_new_cluster()
+        self.create_topic(c1, topic_name, partitions=1)
+        self.create_topic(c2, topic_name, partitions=1)
+
+        # Produce different data to each topic.
+        rpk_c1 = RpkTool(c1)
+        rpk_c2 = RpkTool(c2)
+        num_msgs_per_round = 100
+        msg_len = 1024
+
+        def local_storage_trimmed(cluster: RedpandaService):
+            admin = Admin(cluster)
+            status = admin.get_partition_cloud_storage_status(topic_name, 0)
+            return status["local_log_start_offset"] > 0
+
+        def local_trimmed_or_produce():
+            # Once we trim local storage from both clusters, we're done.
+            if local_storage_trimmed(c1) and local_storage_trimmed(c2):
+                return True
+            for _ in range(num_msgs_per_round):
+                rpk_c1.produce(topic_name, key="1", msg="1" * msg_len)
+                rpk_c2.produce(topic_name, key="2", msg="2" * msg_len)
```

Comment:
nit: produce only to the one that is not trimmed? just to use less resources in case they go askew